juriscraper/pacer/rss_feeds.py

```diff
@@ -98,19 +99,52 @@ def _parse_text(self, text):
 
     @property
     def data(self):
-        """Override this to create a list of docket-like objects instead of the
-         usual dict that is usually provided by the docket report.
+        """Return a list of docket-like objects instead of the usual dict that
+         is usually provided by the BaseDocketReport superclass.
+
+        When CMECF generates the RSS feed, it breaks up items with
+        multiple consecutive entries into multiple RSS items with
+        identical timestamp/id/title.  We reverse that and recombine
+        those items.
         """
         if self._data is not None:
             return self._data
 
         data_list = []
-        for entry in self.feed.entries:
+        for previous_entry, entry, next_entry in previous_and_next(
+                self.feed.entries):
             data = self.metadata(entry)
+
+            # We are guaranteed to only have a single docket entry for each
+            # RSS item, and thus we use data['docket_entries'][0] below.
+            # Coming up with an alternative data representation here and
+            # then transforming it into what CL expects after we're done
+            # iterating over the list is just not worth the bother.
+            data[u'docket_entries'] = self.docket_entries(entry)
+            # BUT: Guarantee this condition persists into the future:
+            assert len(data[u'docket_entries']) <= 1
+
+            # If this entry and the immediately prior entry match
+            # in metadata, then add the current description to
+            # the previous entry's and continue the loop.
+            if (
+                data_list and data_list[-1][u'docket_entries']
+                and data[u'docket_entries']
+                and entry.title == previous_entry.title
+                and entry.link == previous_entry.link
+                and entry.id == previous_entry.id
+                and entry.published == previous_entry.published
+            ):
+                data_list[-1][u'docket_entries'][0][u'short_description'] += (
```

Comment:
I feel like a line like this deserves a comment, if even a short one. Or maybe break the bits into variables for readability?

    previous_entry_desc = data_list[-1][u'docket_entries'][0][u'short_description']
    current_entry_desc = data[u'docket_entries'][0][u'short_description']
    combined_desc = ' AND '.join([previous_entry_desc, current_entry_desc])
    data_list[-1][u'docket_entries'][0][u'short_description'] = combined_desc

That's a bit wordy, but clearer. This multiline thing is pretty bad.

Reply:
I actually don't think that is clearer.
Forcing the repetition of `data_list[-1][u'docket_entries'][0][u'short_description']` is something I would like to avoid, because it requires the reader to parse the 5 levels of hierarchy in their brain twice, and it's not obvious until inspection that they are the same.

Of course, the flipside is that I suspect you have an unfamiliarity with the `+=` operator and do not love it the way you should, and no amount of cajoling will instill that love in you until you have worked with it for a long time. To my eye the avoidance of complex repetition is the strongest case for the so-called "addition assignment" operator. (The real problem is you can't do many of the usual fixes because the complexity is in an _lvalue_ not just an _rvalue_.

And I definitely don't think that `' AND '.join([a, b])` is somehow clearer than `a+' AND '+b`.

I'll think about how to write this better.

At one point I had `prevdata` (or `previous_data`) instead of `data_list[-1]` but I axed it because I felt it wasn't helping. (d59e0072c5a4a539444cfdf1634a372c53901d26)

Reply:
I'm familiar with the `+=`. It's more about how you have three lines of code to do one small thing. That's a smell, but I'll await further approaches.

Reply:
Me:
>> I suspect _A_ and _B_

Mike:
> Not _A_.

So _B_ then. As I feared.

Reply:
So, I suspect you won't go for this one, but it kind of jumped out at me so I thought I'd offer it here:

```py
    def entry(d):
        return d[u'docket_entries'][0]

    def get_append_shorttext(de, text=''):
        de[u'short_description'] += text
        return de[u'short_description']

    previous_entry = entry(data_list[-1])
    current_text = get_append_shorttext(entry(data))
    get_append_shorttext(previous_entry, ' AND ' + current_text)
```

Think of it as a conversation-starter :)

Reply:
Well, while I'm functionally happen with that, it fails PEP-8. And it's actually kind of tricky to fix, and it's why I introduced the helper function, which I agree is totally ridiculous and I'm not sure I could dignify by calling it a strawman:

```py
#        1         2         3         4         5         6         7         8
#2345678901234567890123456789012345678901234567890123456789012345678901234567890
                previous_entry[u'short_description'] += ' AND ' +current_entry[u'short_description']  #1
                previous_entry[u'short_description'] += (  #2
                    ' AND ' + current_entry[u'short_description'])
                previous_entry[u'short_description'] += (  #3
                    ' AND ' +
                    current_entry[u'short_description'])
                previous_entry[u'short_description'] += (  #4
                    ' AND ' +
                    current_entry[u'short_description']
                )
                previous_entry[u'short_description'] += \ 
                    ' AND ' + current_entry[u'short_description']  #5
```

Although I find 5 tempting, I think that backslash terminates are disfavored (although not against PEP8?).
And of 2,3,4 I dunno. I had the sense you wanted to avoid multiple lines for this expression, and yet there's a certain additional clarity. What's your preference?

Reply:
I think 5 has my vote followed closely by 1 (a minor PEP8 violation isn't the end of the world in my book). 2, 3, and 4 just don't seem worth it. I leave the rest of this to you. Let's land this.